### PR TITLE
Makes so turning in place updates riders when riding someone's back

### PR DESCRIPTION
**Pull request by monster860**
this PR is completely untested

## About The Pull Request

Makes so turning in place updates riders when riding someone's back

## Why It's Good For The Game

because you turn and the person riding you doesn't turn and it looks broken

### CODE REVIEW

**Comment by Arturlang** [comment]
Please... test it

**Comment by monster860** [comment]
Testing is for the weak

**Review by Ghommie — APPROVED** [review]

**Comment by IronEleven** [comment]
Not a fan, I and others use it to represent sitting on laps.

File: code/datums/components/riding.dm [hunk 1]
```diff
@@ -47,6 +48,11 @@
 /datum/component/riding/proc/set_vehicle_dir_layer(dir, layer)
 	directional_vehicle_layers["[dir]"] = layer
 
+/datum/component/riding/proc/vehicle_dir_changed(from_dir, to_dir)
+	spawn
```
Comment (Ghommie) [inline code/datums/components/riding.dm:52]: Wait, what the hell?

Comment (monster860) [reply code/datums/components/riding.dm:52]: basically the signal fires *before* the dir is actually changed so uh this is necessary

Comment (Ghommie) [reply code/datums/components/riding.dm:52]: Have you considered adding a "dir override" argument to both procs below and see if that's doable at least?

Comment (silicons) [reply code/datums/components/riding.dm:52]: addtimer's better tbh  if you gotta use this

**Comment by Linzolle** [comment]
I've gotta agree with IronEleven, it can be used for RP tbh.

**Comment by monster860** [comment]
if you want to do lap-sitting you can use pixel-shifting and shizz. The default behavior for this kind of thing should be the one that *doesn't* look broken.

**Pull request merged**